src/app/auth.service.ts

```diff
@@ -32,12 +33,20 @@ export class AuthService {
   private spotifyAuthUrl = 'https://accounts.spotify.com/authorize?';
   private client_id = 'c165fd4b3c454bd5b1c726c172ea2faf';
 
-  constructor(private spotify: SpotifyApiService) { }
+  constructor(
+    private spotify: SpotifyApiService,
+    private localStorage: LocalStorageService
+  ) { }
 
-  public isAuthenticated = () => this.authenticated;
+  public isAuthenticated = () => {
+    return this.authenticated ? this.authenticated : this.checkAuthData();
+  }
   public getAccessToken = () => this.access_token;
 
-  public checkAuthData = (data: any) => {
+  public checkAuthData = (data: any = false) => {
+    if (!data) {
+      data = this._loadFromLocalStorage();
+    }
     if (data.access_token) {
       let test = this._testAccessToken(data.access_token);
```

Comment:
The auth part has a little more work needed. We'll see when I'll get time to handle it.